cdc/processor.go

```diff
@@ -556,177 +554,65 @@ func (p *processor) globalResolvedWorker(ctx context.Context) error {
-	errg, ctx := errgroup.WithContext(ctx)
-
-	errg.Go(func() error {
-		return p.ddlPuller.Run(ctx)
-	})
-
-	errg.Go(func() error {
-		err := p.ddlPuller.CollectRawTxns(ctx, func(ctxInner context.Context, rawTxn model.RawTxn) error {
-			select {
-			case <-ctxInner.Done():
-				return ctxInner.Err()
-			case p.ddlJobsCh <- rawTxn:
-				return nil
-			}
-		})
+		err = p.schemaBuilder.DoGc(changefeedStatus.CheckpointTs)
 		if err != nil {
-			return errors.Annotate(err, "span: ddl")
+			return errors.Trace(err)
 		}
-		return nil
-	})
-	return errg.Wait()
+	}
 }
 
 // syncResolved handle `p.ddlJobsCh` and `p.resolvedTxns`
 func (p *processor) syncResolved(ctx context.Context) error {
-	const bulkLimit = 128
-	var lastResolved model.RawTxn
-	pendingTxns := make([]model.Txn, 0, bulkLimit)
-
-	flush := func(ctx2 context.Context) error {
-		forwardResolved := func() {
-			if lastResolved.Ts > 0 {
-				select {
-				case p.executedTxns <- lastResolved:
-				case <-ctx2.Done():
-				}
-			}
-		}
-		if len(pendingTxns) == 0 {
-			forwardResolved()
-			return nil
-		}
-		if err := p.sink.EmitDMLs(ctx2, pendingTxns...); err != nil {
-			return errors.Trace(err)
-		}
-		log.Info("DMLs Flushed", zap.Any("resolved", lastResolved), zap.Int("bulk", len(pendingTxns)))
-		forwardResolved()
-		txnCounter.WithLabelValues("executed", p.changefeedID, p.captureID).Add(float64(len(pendingTxns)))
-		pendingTxns = pendingTxns[:0]
-		return nil
-	}
-
-	defer close(p.executedTxns)
-
-	flushTick := time.NewTicker(flushDMLsInterval)
-
+	defer log.Info("syncResolved stopped")
 	for {
 		select {
-		case rawTxn, ok := <-p.ddlJobsCh:
-			if !ok {
-				return nil
-			}
-			if rawTxn.IsFake() {
-				atomic.StoreUint64(&p.ddlResolveTS, rawTxn.Ts)
-				continue
-			}
-			t, err := p.mounter.Mount(rawTxn)
+		case row := <-p.output:
+			err := p.sink.EmitRowChangedEvent(ctx, row)
 			if err != nil {
 				return errors.Trace(err)
 			}
-			if !t.IsDDL() {
-				if t.IsDML() {
-					log.Warn("Receive DML txn from ddlJobsCh", zap.Any("txn", t))
-				}
-				continue
-			}
-			p.schemaStorage.AddJob(t.DDL.Job)
-			if err := flush(ctx); err != nil {
-				return errors.Trace(err)
-			}
-			atomic.StoreUint64(&p.ddlResolveTS, rawTxn.Ts)
-		case rawTxn, ok := <-p.resolvedTxns:
-			if !ok {
-				return nil
-			}
-			if rawTxn.IsResolved {
-				lastResolved = rawTxn
-				continue
-			}
-
-			if err := p.schemaStorage.HandlePreviousDDLJobIfNeed(rawTxn.Ts); err != nil {
-				return errors.Trace(err)
-			}
-			txn, err := p.mounter.Mount(rawTxn)
-			if err != nil {
-				return errors.Trace(err)
-			}
-			if p.filter.ShouldIgnoreTxn(&txn) {
-				log.Info("DML txn ignored", zap.Uint64("ts", txn.Ts))
-				continue
-			}
-			p.filter.FilterTxn(&txn)
-			if len(txn.DMLs) == 0 {
-				continue
-			}
-			pendingTxns = append(pendingTxns, txn)
-			if len(pendingTxns) >= bulkLimit {
-				if err := flush(ctx); err != nil {
-					return errors.Trace(err)
-				}
-			}
+			// should filter in sink
```

Comment:
Consider the following strategies, which is better:
1. filter events in the processor, don't emit ignored `RowChangedEvent` to sink
2. filter events in the processor, emit `RowChangedEvent` with an ignorable flag to sink
3. do nothing in the processor, filter event in the sink

Reply:
I think `option 3` is more flexibility, what do you think about it

Reply:
`option 3` seems better
- different sink may have different filter strategy, especially for the DDL compatibility
- option 3 keeps a complete event set